BACKGROUND
==============

Reconstructing the tree of life is a very challenging scientific problem.

**Positioning eukaryotes in the tree of life has been a significant scientific endeavor** 
in evolutionary biology. The slow process of identifying where eukaryotes, the group of complex cellular organisms 
that includes plants, animals, fungi, and many protists, fit within the broader framework of life has required 
resolving deep evolutionary relationships among major lineages of life.

In the 1970s, Carl Woese and his colleagues pioneered the use of ribosomal RNA (rRNA) sequences to infer 
phylogenetic relationships between the known biodiversity. These early attempts using molecular 
markers faced challenges because the deep evolutionary divergence between these domains meant that even highly 
conserved genes had limited resolution power over such vast time scales. Still, the first reconstructions of
the tree of life using rRNA produced the known three-domain system, defining **Bacteria**, **Archaea** and **Eukaryotes**.
These first trees suggested that these three domains were monophyletic, with Eukaryotes being somewhat closely
related to Archaea, and the Root of the tree being somewhere between Bacteria and the Archaea-Eukarya group.

Throughout the 1990s and early 2000s, advancements in molecular techniques and computational phylogenetics 
allowed scientists to move beyond single genes like rRNA and analyze multiple protein-coding genes, improving 
resolution by obtaining **Phylogenomic trees**. However, these analyses often produced conflicting results, 
especially regarding the precise relationship between eukaryotes and archaea. Some studies suggested eukaryotes 
were more closely related to a subgroup of archaea (2-domain tree, where Eukaryotes are inserted within 
the archaeal domain), while others supported a 3-domain tree, where Eukaryotes are more related to Archaea than to
Bacteria, but they do not originate from within the archaeal clade.

In the last 15 years, metagenomics has allowed a rampant discovery of new microbial lineages. 
From 2015 onwards, the discovery of Asgard archaea brought a lot of attention to our understanding of the tree
of life. The Asgard archaea are an archaeal lineage with genes closely resembling those found in eukaryotes. 
This strengthened claims in favour of the 2-domain hypothesis. However, Asgard archaeal genomes were scarce,
limiting the scope of phylogenomic analyses, and several studies produced contradictory results.

In 2020, a [preprint](https://www.biorxiv.org/content/10.1101/2020.10.19.343400v1?versioned=true) was published by an international team in which the authors recovered a large number of
metagenome-assembled genomes of Asgard archaea, thus allowing a reanalysis of the tree of life using
phylogenomics. Here, we take a look at their data and revisit their main conclusions.

